src/temporal_logic/rules.rs

```diff
@@ -710,6 +710,36 @@ pub proof fn leads_to_exists_intro<T, A>(spec: TempPred<T>, a_to_p: spec_fn(A) -
     };
 }
 
+// similar to init_invariant with quantifier
+pub proof fn leads_to_always_exists_intro<T, A>(spec: TempPred<T>, next: ActionPred<T>, p: TempPred<T>, a_to_q: spec_fn(A) -> StatePred<T>)
```

Comment:
Can you name it as "leads_to_exists_stable" and move it next to "leads_to_stable"?